src/libs/utils/creporterutils.cpp

```diff
@@ -206,6 +211,37 @@ QString CReporterUtils::deviceModel()
 #endif
 }
 
+bool CReporterUtils::shouldSavePower()
+{
+#ifndef CREPORTER_UNIT_TEST
+    auto *settings = CReporterPrivacySettingsModel::instance();
+
+    BatteryStatus battery;
+    QEventLoop loop;
+    connect(&battery, &BatteryStatus::initialized, &loop, &QEventLoop::quit);
+    loop.exec();
```

Comment:
Uh. Doing an event loop on getter doesn't feel too good. If we have a general need to have non-async battery status, I wouldn't like to see such code around.

Could be an option to offer some c++ blocking mode or waitForFinished() on the batterystatus, or in libmce-qt which also has battery status class, but as the d-bus interface is so simple, could even consider just doing three blocking d-bus calls here.

Reply:
I wasn't not happy with it either, so if you don't object that I use mce directly... fixed.